Allow executing commands to deal with afk players

nik2143 opened the pull request:
<!--

EssentialsX feature submission guide
====================================

NOTE: Failure to fill out this template properly may result in your PR being
      delayed or ignored without warning.

NOTE: Don't type between any arrows in the template, as this text will be
      hidden. This includes this header block and any other explanation text
      blocks.

Want to discuss your PR before submitting it? Join the EssentialsX Development
server: https://discord.gg/CUN7qVb


EssentialsX is GPL
------------------

By contributing to EssentialsX, you agree to license your code under the
GNU General Public License version 3, which can be found at the link below:
https://github.com/EssentialsX/Essentials/blob/2.x/LICENSE


Instructions
------------

If you are submitting a new feature, please follow the following steps:

1.  Fill out the template in full.
      This includes providing screenshots and a link to the original feature 
      request. If there isn't an existing feature request, we strongly
      recommend opening a new feature request BEFORE opening your PR to
      implement it, as this allows us to review whether we're likely to accept
      your feature in advance, and also allows us to discuss possible
      implementations for the feature. If there is no associated feature
      request, your PR may be delayed or rejected without warning.
      
      You can open a new feature request by following this link:
      https://github.com/EssentialsX/Essentials/issues/new/choose

2.  If you are fixing a performance issue, please use the "Bug fix" PR template
      instead. The "bug fix" template is better suited to performance issues.

3.  Include a demonstration.
      If you are adding commands, please provide screenshots and/or a video
      demonstration of the feature. Similarly, if you are adding a new API,
      please include a link to example code that takes advantage of your
      proposed API. This will aid us in reviewing PRs and speed up the process
      significantly.

-->

### Information

<!--
    Replace #nnnn with the number of the original issue. If this PR implements
    features from multiple issues, you should repeat the phrase "closes #nnnn"
    for each issue. 
-->

This PR closes #2864. 

### Details

**Proposed feature:**  
<!-- Type a description of your proposed feature below this line. -->
This PR allows to execute commands when a player is afk

**Environments tested:**    

<!-- Type the OS you have used below. -->
OS: Windows 11

<!-- Type the JDK version (from java -version) you have used below. -->
Java version: 16.0.1

<!--
    Put an "x" inside the boxes for the server software you have tested this 
    bug fix on. If this feature does not apply to a server, strike through the server software using ~~strikethrough~~. If you have tested on other
    environments, add a new line with relevant details.
-->
- [x] Most recent Paper version (1.XX.Y, git-Paper-BUILD)
- [x] CraftBukkit/Spigot/Paper 1.12.2
- [x] CraftBukkit 1.8.8


**Demonstration:**  
<!--
    Below this block, include screenshots/log snippets from before and after as
    necessary. If you have created or used a test case plugin, please link to a
    download of the plugin, source code and exact version used where possible.
-->

nik2143 committed:
Allow executing commands to deal with afk players

mdcfe commented:
Putting aside the numerous code quality issues in this PR, I'm not sure this feature belongs in EssentialsX. You can already do this with the API. Would be interested in a second opinion from @EssentialsX/maintainers.

mdcfe commented on Essentials/src/main/java/com/earth2me/essentials/ISettings.java:
```diff
@@ -186,7 +186,9 @@
 
     long getAutoAfk();
 
-    long getAutoAfkKick();
```
This breaks API.

mdcfe commented on Essentials/src/main/java/com/earth2me/essentials/ISettings.java:
```diff
@@ -186,7 +186,9 @@
     long getAutoAfk();
 
-    long getAutoAfkKick();
+    long getAutoAfkActionsTimout();
```
Typo

mdcfe commented on Essentials/src/main/java/com/earth2me/essentials/Settings.java:
```diff
@@ -766,7 +767,7 @@ public void reloadConfig() {
         currencySymbol = _getCurrencySymbol();
         worldAliases = _getWorldAliases();
-
+        autoAfkActions = _getAutoAfkActions();
```
Please re-add the blank line between this and the line below.

mdcfe commented on Essentials/src/main/java/com/earth2me/essentials/User.java:
```diff
@@ -804,19 +805,19 @@ public void checkActivity() {
-                if (user.isAuthorized("essentials.kick.notify")) {
-                    user.sendMessage(tl("playerKicked", Console.DISPLAY_NAME, getName(), kickReason));
-                }
+            for (String action : ess.getSettings().getAutoAfkActions()){
```
Please format your code properly.

mdcfe commented on Essentials/src/main/java/com/earth2me/essentials/User.java:
```diff
@@ -804,19 +805,19 @@ public void checkActivity() {
+                action = action.replace("{PLAYER}",getDisplayName());
+                action = action.replace("{USERNAME}",getName());
+                action = action.replace("{WORLDNAME}"
+                        ,getLocation()==null||getLocation().getWorld()==null?"":getLocation().getWorld().getName());
```
Use EssentialsX's existing placeholder system here (you can see an example in the kits code).

mdcfe commented on Essentials/src/main/resources/config.yml:
```diff
@@ -454,11 +454,20 @@ remove-god-on-disconnect: false
+# {USERNAME} - The player's username.
+# {WORLDNAME} - The full name of the player's current world.
+auto-afk-actions:
+  - "kick {USERNAME}"
```
This change means players will no longer see the configured AFK kick message. AFK kicks should be kept separate from other AFK timeout actions to avoid breaking existing setups.

I'm also not convinced that running all commands at once is a good design for this feature. It would be far more useful to allow setting different timeouts for each action.

nik2143 committed:
Restore Auto Afk Kick

Restore the auto afk kick

nik2143 committed:
Change Auto Afk Actions

Allows to set actions to execute at every timeout
Change placeholders to existing placeholder system

nik2143 committed:
Merge branch '2.x' into afk-commands

nik2143 committed:
Merge branch '2.x' into afk-commands

nik2143 committed:
Merge branch '2.x' into afk-commands

nik2143 committed:
Merge branch '2.x' into afk-commands
